src/table.rs

```diff
@@ -463,6 +463,14 @@ impl ValueTable {
 					},
 				}
 			}
+
+			if buf.offset() > entry_end {
+				return Err(crate::error::Error::Corruption(format!(
+					"Unexpected entry size. Expected at least {} bytes",
+					entry_end
```

Comment:
```suggestion
					buf.offset() - 2
```
2 assume we only read size.